cli/cloe_launch/exec.py

```diff
@@ -382,80 +389,136 @@ def _read_anonymous_profile(self, conanfile) -> None:
         logging.debug(f"Write: {prompt_sh_file}")
         with prompt_sh_file.open("w") as file:
             file.write(prompt_sh_data)
 
     def _write_bashrc(self) -> None:
         """Write .bashrc file.
 
-        When creating a new shell, it's not possible to just "source" a file
-        and continue with the normal interactive session. So we need to create
-        a bashrc file that will be used instead, which loads the normal
-        configuration, and then augments it with the extra variables we need.
+        When launching a shell, this file is used as the Bash user configuration.
+        This is necessary to properly override the prompt with [cloe-shell] prefix.
         """
         bashrc_file = self.runtime_dir / ".bashrc"
-        bashrc_data = """\
-        source /etc/bash.bashrc
-        if [ -f ~/.bashrc ]; then
-            source ~/.bashrc
-        fi
-        OLD_PS1="$PS1"
-        source "$(dirname "$BASH_SOURCE[0]")/launcher_env.sh"
-        PS1="$OLD_PS1"
-        source "$(dirname "$BASH_SOURCE[0]")/prompt.sh"
-        """
+        bashrc_data = textwrap.dedent("""\
+            source /etc/bash.bashrc
+            if [ -f ~/.bashrc ]; then
+                source ~/.bashrc
+            fi
+            OLD_PS1="$PS1"
+            source "$(dirname "$BASH_SOURCE[0]")/activate_all.sh"
+            PS1="$OLD_PS1"
+            source "$(dirname "$BASH_SOURCE[0]")/prompt.sh"
+            """)
         logging.debug(f"Write: {bashrc_file}")
         with bashrc_file.open("w") as file:
             file.write(bashrc_data)
 
     def _write_zshrc(self) -> None:
-        """Write .zshrc file."""
-        zshrc_file = self.runtime_dir / ".zshrc"
-        zshrc_data = f"""\
-        ZDOTDIR="${{OLD_ZDOTDIR:-$HOME}}"
-        source "${{ZDOTDIR}}/.zshenv"
-        source "${{ZDOTDIR}}/.zshrc"
-        OLD_PS1="$PS1"
-        source "{self.runtime_dir}/launcher_env.sh"
-        PS1="$OLD_PS1"
-        source "{self.runtime_dir}/prompt.sh"
+        """Write .zshrc file.
+
+        When launching a shell, this file is used as the Zsh user configuration.
+        This is necessary to properly override the prompt with [cloe-shell] prefix.
         """
+        zshrc_file = self.runtime_dir / ".zshrc"
+        zshrc_data = textwrap.dedent(f"""\
+            ZDOTDIR="${{OLD_ZDOTDIR:-$HOME}}"
+            source "${{ZDOTDIR}}/.zshenv"
+            source "${{ZDOTDIR}}/.zshrc"
+            OLD_PS1="$PS1"
+            source "{self.runtime_dir}/activate_all.sh"
+            PS1="$OLD_PS1"
+            source "{self.runtime_dir}/prompt.sh"
+            """)
         logging.debug(f"Write: {zshrc_file}")
         with zshrc_file.open("w") as file:
             file.write(zshrc_data)
 
+    def _write_cloe_env(self) -> None:
+        """Derive important CLOE_ variables and write environment_cloe.sh.env file."""
+        env = Environment(self.runtime_dir / "activate_run.sh", source_file=True)
+        if env.has("CLOE_SHELL"):
```

Comment:
This check doesn't work since the `"CLOE_SHELL"` is not preserved.

I am not sure whether your intention is that this environment variable shall be part of Environment._preserve list or if this is not the right place to call for this check. 

So maybe add `"CLOE_SHELL"` to  Environment._preserve list?

Reply:
Great find! Implemented a fix.